src/main.c

```diff
@@ -54,6 +58,10 @@ int main(int argc, char* argv[])
     // Parse options
     parse_options(argc, argv);
 
+#ifdef __CYGWIN__
+    windows_socket_initialize();
```

Comment:
This is required to make liblxi work on windows?

If so, this should be moved to lxi_init() in liblxi.

Reply:
This makes lxi cli to work. Lxi gui already work. 
I also though to move this in liblxi. But I am not shure what impact can it make on gui which already initializez socket (probably some library initializes)

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsastartup

The documentation says:

> Once an application or DLL has made a successful WSAStartup call, it can proceed to make other Windows Sockets calls as needed. When it has finished using the services of the Winsock DLL, the application must call [WSACleanup](https://learn.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-wsacleanup) to allow the Winsock DLL to free internal Winsock resources used by the application.
>
> An application can call WSAStartup more than once if it needs to obtain the [WSADATA](https://learn.microsoft.com/en-us/windows/desktop/api/winsock/ns-winsock-wsadata) structure information more than once. On each such call, the application can specify any version number supported by the Winsock DLL.

Which makes me think we it would be okay to put it in the library.

However, then it also says:
> The WSAStartup function typically leads to protocol-specific helper DLLs being loaded. As a result, the WSAStartup function should not be called from the DllMain function in a application DLL. This can potentially cause deadlocks. For more information, please see the [DLL Main Function](https://learn.microsoft.com/en-us/windows/win32/dlls/dllmain).

This is a very poorly designed interface. When using any sort of high communications library, the application should not have any knowledge of WSAStartup() but it seems Microsoft has demanded it so.

The idea of liblxi is to abstract everything required to do lxi communication, and in case of cygwin I think maybe we should just run the risk and move WSAStartup() to lxi_init(). Perhaps introduce a lxi_deinit() for WSACleanup().

Reply:
Agree, liblxi should take care of communication protocols. I will move this to liblxi. Hope gui won't have stabillity issues.

Reply:
close this pull request because functionality will move in liblxi